app/ui-react/packages/ui/src/Customization/apiClientConnectors/detail/ApiConnectorDetailConfigEdit.tsx

```diff
@@ -56,68 +58,92 @@ export const ApiConnectorDetailConfigEdit: React.FunctionComponent<IApiConnector
   return (
     <Form isHorizontal={true} data-testid={'api-connector-details-form'}>
       {i18nRequiredText}
-      <FormGroup
-        label={i18nLabelName}
-        isRequired={true}
-        fieldId="connector-name"
-        helperTextInvalid={i18nNameHelper}
-        validated={isValid}
-      >
-        <TextInput
-          value={properties.name}
+      {properties.name && (
+        <FormGroup
+          label={i18nLabelName}
           isRequired={true}
-          type="text"
-          id="connector-name"
-          aria-describedby="horizontal-form-name-helper"
-          data-testid={'api-connector-name-field'}
-          name="name"
-          onChange={onChange}
+          fieldId="connector-name"
+          helperTextInvalid={i18nNameHelper}
           validated={isValid}
-        />
-      </FormGroup>
-      <FormGroup 
-        label={i18nLabelDescription} 
-        fieldId="connector-description">
-        <TextArea
-          value={properties.description}
-          onChange={onChange}
-          data-testid={'api-connector-description-field'}
-          name="description"
-          id="connector-description"
-        />
-      </FormGroup>
-      <FormGroup
-        label={i18nLabelHost}
-        isRequired={false}
-        fieldId="connector-host"
-      >
-        <TextInput
-          value={properties.host}
+        >
+          <TextInput
+            value={properties.name}
+            isRequired={true}
+            type="text"
+            id="connector-name"
+            aria-describedby="horizontal-form-name-helper"
+            data-testid={'api-connector-name-field'}
+            name="name"
+            onChange={onChange}
+            validated={isValid}
+          />
+        </FormGroup>
+      )}
+      {properties.description && (
```

Comment:
What a pain that the formatting has caused the diff to look like this.. all I did was add a check for the property like `{properties.description && ( // display the field )}`, and of course add a form group for the Address property.

Reply:
np :smile: